tests/unit/template_cassdc_test.go

```diff
@@ -725,7 +730,8 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
 			verifyMedusaVolumeMounts(medusaContainer)
 
 			Expect(len(cassdc.Spec.PodTemplateSpec.Spec.Volumes)).To(Equal(3))
-			Expect(cassdc.Spec.PodTemplateSpec.Spec.Volumes[0].Name).To(Equal(medusaConfigVolumeName))
+			Expect(cassdc.Spec.PodTemplateSpec.Spec.Volumes[0].Name).To(Equal(CassandraConfigVolumeName))
```

Comment:
We should update the volume checks so that they do not rely on order, similar to what we do with containers. It will make the tests less brittle. 

Note that we do care about order for initContainers though.